src/cli/cli.cpp

```diff
@@ -276,11 +276,33 @@ template <> otError Interpreter::Process<Cmd("version")>(Arg aArgs[])
 
 template <> otError Interpreter::Process<Cmd("reset")>(Arg aArgs[])
 {
-    OT_UNUSED_VARIABLE(aArgs);
+    otError error = OT_ERROR_NONE;
 
-    otInstanceReset(GetInstancePtr());
+    if (aArgs[0].IsEmpty())
+    {
+        otInstanceReset(GetInstancePtr());
+    }
 
-    return OT_ERROR_NONE;
+#if OPENTHREAD_CONFIG_PLATFORM_BOOTLOADER_MODE_ENABLE
+    /**
+     * @cli reset bootloader
+     * @code
+     * reset bootloader
+     * @endcode
+     * @par
+     * Resets platform to bootloader mode, if supported.
```

Comment:
```suggestion
     * @par api_copy
     * #otInstanceResetToBootloader
```

Reply:
This will use the API description and link to it when we import into openthread.io, rather than hard-coding another description.